clients/src/main/java/org/apache/kafka/common/protocol/Protocol.java

```diff
@@ -103,14 +103,48 @@ private static void schemaToFieldTableHtml(Schema schema, StringBuilder b) {
         b.append("<th>Description</th>\n");
         b.append("</tr>");
         for (BoundField field : fields) {
-            b.append("<tr>\n");
-            b.append("<td>");
-            b.append(field.def.name);
-            b.append("</td>");
-            b.append("<td>");
-            b.append(field.def.docString);
-            b.append("</td>");
-            b.append("</tr>\n");
+            if (field.def.type instanceof TaggedFields) {
+                TaggedFields taggedFields = (TaggedFields) field.def.type;
+                // Only include the field in the table if there are actually tags defined
+                if (taggedFields.numFields() > 0) {
```

Comment:
I wonder if we should still keep a row in the table even if there are no tagged fields. Otherwise in some cases we end up with TAG_BUFFER appearing in the schema and missing in the table. This can be confusing if there are multiple TAG_BUFFER.
For example:
<img width="1011" alt="image" src="https://github.com/user-attachments/assets/b887e2cf-678c-47c2-98c3-67d9ebcc63ef">

Could we also make the names match? In the schema we use `TAG_BUFFER` but then in the table we use `_tagged_fields`.

Reply:
<img width="997" alt="Screenshot 2024-10-22 at 15 35 42" src="https://github.com/user-attachments/assets/95209248-4e32-4932-b5c0-3da3f09786f7">

I've added back the empty rows for situations in which there are no tagged fields defined, and also renamed "TAG_BUFFER" as "_tagged_fields".